src/envs/screws.py

```diff
@@ -0,0 +1,482 @@
+        gripper_y = state.get(self._robot, "pose_y")
+        plt.plot([(gripper_x - self.gripper_width / 2.0) + self._screw_width,
+                  (gripper_x + self.gripper_width / 2.0) + self._screw_width],
+                 [gripper_y, gripper_y],
+                 color="black")
+        plt.plot([(gripper_x - self.gripper_width / 2.0) + self._screw_width,
+                  (gripper_x - self.gripper_width / 2.0) + self._screw_width],
+                 [gripper_y - self.gripper_width / 4.0, gripper_y],
+                 color="black")
+        plt.plot([(gripper_x + self.gripper_width / 2.0) + self._screw_width,
+                  (gripper_x + self.gripper_width / 2.0) + self._screw_width],
+                 [gripper_y - self.gripper_width / 4.0, gripper_y],
+                 color="black")
+
+        # Find which object is the goal screw.
+        assert len(task.goal) == 1
+        goal_pred = list(task.goal)[0]
+        goal_screw = goal_pred.objects[0]
+
+        # Draw screws.
+        for screw in state.get_objects(self._screw_type):
+            screw_x = state.get(screw, "pose_x")
+            screw_y = state.get(screw, "pose_y")
+            if screw == goal_screw:
+                color = 'gold'
+            else:
+                color = 'grey'
+            rect = plt.Rectangle((screw_x - self._screw_width / 2.0,
+                                  screw_y - self._screw_height / 2.0),
+                                 self._screw_width,
+                                 self._screw_height,
+                                 edgecolor=color,
+                                 facecolor=color)
+            ax.add_patch(rect)
+
+        img = utils.fig2data(fig)
+        plt.close()
+        return [img]
+
+    @property
+    def predicates(self) -> Set[Predicate]:
+        return {
+            self._GripperCanPickScrew, self._AboveReceptacle,
+            self._HoldingScrew, self._ScrewInReceptacle
+        }
+
+    @property
+    def goal_predicates(self) -> Set[Predicate]:
+        return {self._ScrewInReceptacle}
+
+    @property
+    def types(self) -> Set[Type]:
+        return {self._screw_type, self._gripper_type, self._receptacle_type}
+
+    @property
+    def options(self) -> Set[ParameterizedOption]:
+        return {
+            self._MoveToScrew, self._MoveToReceptacle, self._MagnetizeGripper,
+            self._DemagnetizeGripper
+        }
+
+    @property
+    def action_space(self) -> Box:
+        # dimensions: [dx, dy, magnetized vs. unmagnetized].
+        lowers = np.array([-self._rz_width, -self._rz_height, 0.0],
+                          dtype=np.float32)
+        uppers = np.array([self._rz_width, self._rz_height, 1.0],
+                          dtype=np.float32)
+        return Box(lowers, uppers)
+
+    def simulate(self, state: State, action: Action) -> State:
+        assert self.action_space.contains(action.arr)
+        dx, dy, magnetization = action.arr
+        if dx != 0.0 or dy != 0.0:
```

Comment:
would anything be lost if we _always_ transition move, and then also transition (de)magnetize? if we ever try to learn options in this environment, the != 0.0 thing here is going to be an issue

Reply:
yup - we can definitely do this and not lose anything! I'll go ahead and make this change